File: CarSpecAngular/carspec/src/app/core/models/models/transmissionDto.ts
export class CreateTransmissionDto {
  transmissionType: string;
  numberOfGears: number;

  constructor(
    transmissionType: string,
    numberOfGears: number
  ) {
    this.transmissionType = transmissionType;
    this.numberOfGears = numberOfGears;
  }
}

export class TransmissionDto {
  transmissionId: number;
  transmissionType: string;
  numberOfGears: number;

  constructor(
    transmissionId: number,
    transmissionType: string,
    numberOfGears: number
  ) {
    this.transmissionId = transmissionId;
    this.transmissionType = transmissionType;
    this.numberOfGears = numberOfGears;
  }
}

File: CarSpecAPI/Data/Models/DataExtractionModel/TransmissionDto.cs
namespace CarSpecAPI.Data.Models.DataExtractionModel
{
    public sealed class TransmissionDto
    {
        public int IndexId { get; set; }
        public string TransmissionRef { get; set; } = null!;
        public string? TransmissionType { get; set; }
        public int? NumberOfGears { get; set; }
        public bool? HasManualOverride { get; set; }
        public bool? HasPaddleShifters { get; set; }
    }
}
